pkg/server/auth_middleware.go

```diff
@@ -22,12 +25,19 @@ func (s *Server) AuthMiddleware() gin.HandlerFunc {
 			c.Next()
 			return
 		}
-		for _, k := range s.authInfo {
-			if k.IdentityToken == amr.IdentityToken {
-				c.Set("identity_token", amr.IdentityToken)
-				c.Next()
-				return
+		_, err := s.Queries.GetUser(context.Background(), amr.IdentityToken)
+		if err != nil {
+			if errors.Is(err, pgx.ErrNoRows) {
+				respondWithError(c, http.StatusUnauthorized,
```

Comment:
This function requires an explicit `c.Next()` and `return`. Otherwise, it will continue.

See https://github.com/tensorchord/envd-server/pull/42#discussion_r1006625870